s3/libirods_s3.hpp

```diff
@@ -171,6 +173,8 @@ S3Protocol s3GetProto( irods::plugin_property_map& _prop_map);
 
 S3STSDate s3GetSTSDate( irods::plugin_property_map& _prop_map);
 
+long s3GetMaxUploadSizeMB (irods::plugin_property_map& _prop_map);
```

Comment:
moving forward, we should avoid types like `long` as they don't convey enough information.

we should use types like `std::int64_t` as they help to enforce the range of values we expect to handle.

that can be handled in a separate PR.

Reply:
Wrote an issue on this for future work.

Reply:
link to it... #2046